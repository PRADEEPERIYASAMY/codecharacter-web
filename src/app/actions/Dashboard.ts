import { SidePanelTab } from 'app/reducers/Dashboard';
import { action } from 'typesafe-actions';

export namespace DashboardActions {
  export enum Type {
    SET_SIDE_PANEL_TAB = 'SET_SIDE_PANEL_TAB',
    RESET_DASHBOARD_STATE = 'RESET_DASHBOARD_STATE',
  }

  export const setSidePanelTab = (sidePanelTab: SidePanelTab) =>
    action(Type.SET_SIDE_PANEL_TAB, {
      sidePanelTab,
    });

  export const resetDashboardState = () => action(Type.RESET_DASHBOARD_STATE);
}
