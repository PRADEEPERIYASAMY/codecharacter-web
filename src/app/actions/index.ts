export * from 'app/actions/dashboard';
