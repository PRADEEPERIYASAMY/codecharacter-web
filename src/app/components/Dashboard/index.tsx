import * as React from 'react';
import * as style from './index.css';
import SplitPane from 'react-split-pane';
import { Editor } from './Code/editor';

export namespace Dashboard {
  export interface Props {}

  export interface State {
    code: String;
  }
}

export class Dashboard extends React.Component<Dashboard.Props, Dashboard.State> {
  constructor(props: Dashboard.Props, context?: any) {
    super(props, context);
    this.state = {
      code: ''
    };
  }

  render() {
    return (
      <SplitPane split="vertical" minSize={50} defaultSize={100} resizerClassName={style.vertical}>
        <Editor />
        <SplitPane split="horizontal" resizerClassName={style.horizontal}>
          <div />
          <div />
        </SplitPane>
      </SplitPane>
    );
  }
}
