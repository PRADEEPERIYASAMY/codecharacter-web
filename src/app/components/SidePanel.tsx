import CommitLog from 'app/containers/code/CommitLog';
import EditorSettings from 'app/containers/code/EditorSettings';
import Leaderboard from 'app/containers/Leaderboard';
import { SidePanelTab } from 'app/reducers/Dashboard';
import * as React from 'react';

export class SidePanel extends React.Component<SidePanel.Props, {}> {
  public render() {
    const { sidePanelTab, sidePanelWidth } = this.props;
    let sidePanel = <div className="SidePanel" />;
    switch (sidePanelTab) {
      case SidePanelTab.NONE:
        sidePanel = <div className="SidePanel" />;
        break;
      case SidePanelTab.EDITOR_SETTINGS:
        sidePanel = (
          <div
            className="SidePanel"
            style={{
              backgroundColor: '#1c1c1c',
              height: '100vh',
              width: `${sidePanelWidth}px`,
            }}
          >
            <EditorSettings />
          </div>
        );
        break;
      case SidePanelTab.LEADERBOARD:
        sidePanel = (
          <div
            className="SidePanel"
            style={{
              backgroundColor: '#1c1c1c',
              height: '100vh',
              width: `${sidePanelWidth}px`,
            }}
          >
            <Leaderboard />
          </div>
        );
        break;
      case SidePanelTab.COMMIT_LOG:
        sidePanel = (
          <div
            className="SidePanel"
            style={{
              backgroundColor: '#1c1c1c',
              height: '100vh',
              width: `${sidePanelWidth}px`,
            }}
          >
            <CommitLog />
          </div>
        );
        break;
    }
    return sidePanel;
  }
}

export namespace SidePanel {
  export interface StateProps {
    sidePanelTab: SidePanelTab;
  }

  export interface OwnProps {
    sidePanelWidth: number;
  }

  export type Props = StateProps & OwnProps;
}
