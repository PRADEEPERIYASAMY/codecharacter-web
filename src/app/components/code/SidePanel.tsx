import EditorSettings from 'app/containers/code/EditorSettings';
import { SidePanelTab } from 'app/reducers/Dashboard';
import * as React from 'react';

// tslint:disable-next-line:variable-name
export class SidePanel extends React.Component<SidePanel.Props, {}> {
  public componentWillReceiveProps(nextProps: SidePanel.Props) {
    const { sidePanelTab, onShowSidePanel, onHideSidePanel } = nextProps;
    if (sidePanelTab === 'EDITOR_SETTINGS') {
      onShowSidePanel();
    } else {
      onHideSidePanel();
    }
  }
  public render() {
    const { sidePanelTab } = this.props;
    let sidePanel = (
      <div
        className="SidePanel"
        style={{
          backgroundColor: '#1c1c1c',
          height: '100vh',
        }}
      />
    )
    switch (sidePanelTab) {
      case SidePanelTab.NONE:
        sidePanel = (
          <div
            className="SidePanel"
            style={{
              backgroundColor: '#1c1c1c',
              height: '100vh',
            }}
          />
        )
        break;
      case SidePanelTab.EDITOR_SETTINGS:
        sidePanel = (
          <EditorSettings/>
        )
        break;
      default:
        sidePanel = (
          <div
            className="SidePanel"
            style={{
              backgroundColor: '#1c1c1c',
              height: '100vh',
            }}
          />
        )
    }
    return sidePanel
  }
}

export namespace SidePanel {

  export interface OwnProps {
    onShowSidePanel: () => void
    onHideSidePanel: () => void
  }

  export interface StateProps {
    sidePanelTab: SidePanelTab;
  }

  export type Props = OwnProps & StateProps & {};
}
