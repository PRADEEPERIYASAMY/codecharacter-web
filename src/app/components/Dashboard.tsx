import Editor from 'app/containers/code/Editor';
import GameLog from 'app/containers/code/GameLog';
import Sidebar from 'app/containers/code/Sidebar';
import SidePanel from 'app/containers/code/SidePanel';
import * as style from 'app/styles/Dashboard.css';
import * as React from 'react';
import { Grid, Row } from 'react-bootstrap';
/* tslint:disable-next-line:import-name */
import SplitPane from 'react-split-pane';

export class Dashboard extends React.Component<{}, Dashboard.State> {
  constructor(props: {}) {
    super(props);
    this.state = {
      editorWidth: 675,
      leftPartitionWidth: 725,
      usedUpLeftPartitionWidth: 50,
    };
  }

  public render() {
    const { editorWidth, leftPartitionWidth } = this.state;

    return (
      <div>
        <SplitPane
          split="vertical"
          minSize={500}
          defaultSize={725}
          size={leftPartitionWidth}
          resizerClassName={style.vertical}
          onChange={this.onResize}
        >
          <Grid fluid={true}>
            <Row>
              <div
                style={{
                  height: '100vh',
                }}
              >
                <Sidebar />
              </div>
              <SidePanel
                onShowSidePanel={this.onShowSidePanel}
                onHideSidePanel={this.onHideSidePanel}
              />
              <div
                style={{
                  height: '100vh',
                }}
              >
                <Editor editorWidth={editorWidth} />
              </div>
            </Row>
          </Grid>
          <SplitPane split="horizontal" defaultSize={300} resizerClassName={style.horizontal}>
            <div />
            <GameLog />
          </SplitPane>
        </SplitPane>
      </div>
    );
  }

  private onResize = (size: number): void => {
    this.setState({
      editorWidth: size - this.state.usedUpLeftPartitionWidth,
      leftPartitionWidth: size,
    });
  };

  /* Sidebar width: 50px, Settings width: 350px, force update and rerender Editor by updating editorWidth */
  private onShowSidePanel = (): void => {
    let  availableWidth = window.innerWidth - this.state.usedUpLeftPartitionWidth
    const usedUpLeftPartitionWidth = 400
    const editorRatio = this.state.editorWidth / availableWidth;
    availableWidth = window.innerWidth - usedUpLeftPartitionWidth
    const editorWidth = Math.floor(editorRatio * availableWidth)
    const leftPartitionWidth = usedUpLeftPartitionWidth + editorWidth

    this.setState({
      editorWidth,
      leftPartitionWidth,
      usedUpLeftPartitionWidth
    })
  };

  private onHideSidePanel = (): void => {
    let  availableWidth = window.innerWidth - this.state.usedUpLeftPartitionWidth
    const usedUpLeftPartitionWidth = 50
    const editorRatio = this.state.editorWidth / availableWidth;
    availableWidth = window.innerWidth - usedUpLeftPartitionWidth
    const editorWidth = Math.floor(editorRatio * availableWidth)
    const leftPartitionWidth = usedUpLeftPartitionWidth + editorWidth

    this.setState({
      editorWidth,
      leftPartitionWidth,
      usedUpLeftPartitionWidth
    })
  };
}
export namespace Dashboard {
  export interface State {
    editorWidth: number;
    leftPartitionWidth: number;
    usedUpLeftPartitionWidth: number;
  }
}
