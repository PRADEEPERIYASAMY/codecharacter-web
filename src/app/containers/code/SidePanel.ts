import { DashboardActions } from 'app/actions';
import { SidePanel } from 'app/components/code/SidePanel';
import { RootState } from 'app/reducers';
import { connect } from 'react-redux';

const mapStateToProps = (rootState: RootState) => {
  return {
    sidePanelTab: rootState.dashboard.sidePanelTab,
  };
};

const sidePanelContainer = connect<SidePanel.StateProps, {}, SidePanel.OwnProps>(
  mapStateToProps,
  {
    setSidePanelTab: DashboardActions.setSidePanelTab,
  },
)(SidePanel);

export default sidePanelContainer;
