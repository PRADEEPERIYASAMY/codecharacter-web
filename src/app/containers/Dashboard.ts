import { Dashboard } from 'app/components/Dashboard';
import { RootState } from 'app/reducers';
import { SidePanelTab } from 'app/reducers/Dashboard';
import { connect } from 'react-redux';

const mapStateToProps = (rootState: RootState) => {
  return {
    sidePanelOpen: rootState.dashboard.sidePanelTab !== SidePanelTab.NONE,
  };
};

const dashboardContainer = connect<Dashboard.StateProps, {}, {}>(mapStateToProps)(Dashboard);

export default dashboardContainer;
