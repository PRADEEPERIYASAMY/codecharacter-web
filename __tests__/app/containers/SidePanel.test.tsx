import { DashboardActions } from 'app/actions';
import SidePanel from 'app/containers/code/SidePanel';
import { SidePanelTab } from 'app/reducers/Dashboard';
import { configureStore } from 'app/store';
import { mount } from 'enzyme';
import * as React from 'react';
// import { RootState } from 'app/reducers';

describe ('SidePanel Container', () => {
  const store = configureStore();

  it ('Should render empty <div/> initially', () => {
    const wrapper = mount(<SidePanel
      onShowSidePanel={Function}
      onHideSidePanel={Function}
    />, {
          context: {
            store,
          }
      });
      expect(wrapper.find('.SidePanel').children()).toHaveLength(0)
  });

  it('Should render', () => {
      store.dispatch(DashboardActions.setSidePanelTab(SidePanelTab.EDITOR_SETTINGS))
      const wrapper = mount(<SidePanel
        onShowSidePanel={Function}
        onHideSidePanel={Function}
      />, {
            context: {
                store,
            }
        });
        expect(wrapper).toMatchSnapshot();
    });

    it('Should change font', () => {
      const wrapper = mount(<SidePanel
        onShowSidePanel={Function}
        onHideSidePanel={Function}/>, {
            context: {
                store,
            }
        });

        expect(wrapper.find('select.font-size-control').props().value).toBe(16);
        wrapper.find('select.font-size-control').simulate('change', { target: { value: '32' } });
        expect(wrapper.find('select.font-size-control').props().value).toBe(32);
    });

    it('Should change theme', () => {
      const wrapper = mount(<SidePanel
        onShowSidePanel={Function}
        onHideSidePanel={Function}/>, {
            context: {
                store,
            }
        });

        expect(wrapper.find('select.theme-control').props().value).toBe('monokai');
        wrapper.find('select.theme-control').simulate('change', { target: { value: 'github' } });
        expect(wrapper.find('select.theme-control').props().value).toBe('github');
    });


  it('Should render empty <div/> after sidePanel toggle off', () => {
    store.dispatch(DashboardActions.setSidePanelTab(SidePanelTab.NONE))
      const wrapper = mount(<SidePanel
        onShowSidePanel={Function}
        onHideSidePanel={Function}
      />, {
            context: {
              store,
            }
        });
        expect(wrapper.find('.SidePanel').children()).toHaveLength(0)
    });

});
