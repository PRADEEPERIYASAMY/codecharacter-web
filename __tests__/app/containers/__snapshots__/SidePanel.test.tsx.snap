// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SidePanel Container Should render 1`] = `
<Connect(SidePanel)
  onHideSidePanel={[Function]}
  onShowSidePanel={[Function]}
>
  <SidePanel
    onHideSidePanel={[Function]}
    onShowSidePanel={[Function]}
    setSidePanelTab={[Function]}
    sidePanelTab="EDITOR_SETTINGS"
  >
    <Connect(EditorSettings)>
      <EditorSettings
        basicAutoCompletion={false}
        changeFontSize={[Function]}
        changeTheme={[Function]}
        enableAutoCompletion={[Function]}
        enableSnippets={[Function]}
        fontSize={16}
        snippets={false}
        theme="monokai"
      >
        <div
          className="SidePanel"
          style={
            Object {
              "backgroundColor": "#1c1c1c",
              "height": "100vh",
            }
          }
        >
          <Grid
            bsClass="container"
            className=""
            componentClass="div"
            fluid={true}
          >
            <div
              className="container-fluid"
            >
              <Row
                bsClass="row"
                className="justify-content-between py-2 pl-3"
                componentClass="div"
              >
                <div
                  className="justify-content-between py-2 pl-3 row"
                >
                  <Col
                    bsClass="col"
                    className="text-light font-weight-bold my-auto"
                    componentClass="div"
                  >
                    <div
                      className="text-light font-weight-bold my-auto"
                    >
                      SETTINGS
                    </div>
                  </Col>
                </div>
              </Row>
              <Row
                bsClass="row"
                componentClass="div"
              >
                <div
                  className="row"
                >
                  <Col
                    bsClass="col"
                    className="mb-1"
                    componentClass="div"
                    sm={12}
                  >
                    <div
                      className="mb-1 col-sm-12"
                    >
                      <div
                        className="font-weight-bold py-1"
                      >
                        Font Size
                      </div>
                      <p>
                        The size of text in the code editor
                      </p>
                      <FormGroup
                        bsClass="form-group"
                        controlId="fontSize"
                      >
                        <div
                          className="form-group"
                        >
                          <select
                            className="form-control font-size-control"
                            onChange={[Function]}
                            value={16}
                          >
                            <option
                              key="8"
                              value={8}
                            >
                              8
                            </option>
                            <option
                              key="10"
                              value={10}
                            >
                              10
                            </option>
                            <option
                              key="12"
                              value={12}
                            >
                              12
                            </option>
                            <option
                              key="14"
                              value={14}
                            >
                              14
                            </option>
                            <option
                              key="16"
                              value={16}
                            >
                              16
                            </option>
                            <option
                              key="18"
                              value={18}
                            >
                              18
                            </option>
                            <option
                              key="20"
                              value={20}
                            >
                              20
                            </option>
                            <option
                              key="22"
                              value={22}
                            >
                              22
                            </option>
                            <option
                              key="24"
                              value={24}
                            >
                              24
                            </option>
                            <option
                              key="26"
                              value={26}
                            >
                              26
                            </option>
                            <option
                              key="28"
                              value={28}
                            >
                              28
                            </option>
                            <option
                              key="30"
                              value={30}
                            >
                              30
                            </option>
                            <option
                              key="32"
                              value={32}
                            >
                              32
                            </option>
                            <option
                              key="34"
                              value={34}
                            >
                              34
                            </option>
                            <option
                              key="36"
                              value={36}
                            >
                              36
                            </option>
                            <option
                              key="38"
                              value={38}
                            >
                              38
                            </option>
                            <option
                              key="40"
                              value={40}
                            >
                              40
                            </option>
                          </select>
                        </div>
                      </FormGroup>
                    </div>
                  </Col>
                  <Col
                    bsClass="col"
                    className="mb-1"
                    componentClass="div"
                    sm={12}
                  >
                    <div
                      className="mb-1 col-sm-12"
                    >
                      <div
                        className="font-weight-bold py-1"
                      >
                        Editor Theme
                      </div>
                      <p>
                        The syntax and overall theme of the code editor
                      </p>
                      <FormGroup
                        bsClass="form-group"
                        controlId="theme"
                      >
                        <div
                          className="form-group"
                        >
                          <select
                            className="form-control theme-control"
                            onChange={[Function]}
                            value="monokai"
                          >
                            <option
                              key="monokai"
                              value="monokai"
                            >
                              monokai
                            </option>
                            <option
                              key="github"
                              value="github"
                            >
                              github
                            </option>
                            <option
                              key="tomorrow"
                              value="tomorrow"
                            >
                              tomorrow
                            </option>
                            <option
                              key="kuroir"
                              value="kuroir"
                            >
                              kuroir
                            </option>
                            <option
                              key="twilight"
                              value="twilight"
                            >
                              twilight
                            </option>
                            <option
                              key="xcode"
                              value="xcode"
                            >
                              xcode
                            </option>
                            <option
                              key="textmate"
                              value="textmate"
                            >
                              textmate
                            </option>
                            <option
                              key="solarized_dark"
                              value="solarized_dark"
                            >
                              solarized_dark
                            </option>
                            <option
                              key="solarized_light"
                              value="solarized_light"
                            >
                              solarized_light
                            </option>
                            <option
                              key="terminal"
                              value="terminal"
                            >
                              terminal
                            </option>
                          </select>
                        </div>
                      </FormGroup>
                    </div>
                  </Col>
                  <Col
                    bsClass="col"
                    className="mb-1"
                    componentClass="div"
                    sm={12}
                  >
                    <div
                      className="mb-1 col-sm-12"
                    >
                      <div
                        className="font-weight-bold py-1"
                      >
                        Auto Completion
                      </div>
                      <p>
                        Use basic auto code completion in the editor
                      </p>
                      <FormGroup
                        bsClass="form-group"
                        controlId="autoCompletion"
                      >
                        <div
                          className="form-group"
                        >
                          <select
                            className="form-control auto-complete-control"
                            onChange={[Function]}
                            value="disabled"
                          >
                            <option
                              key="enabled"
                              value="enabled"
                            >
                              Enabled
                            </option>
                            <option
                              key="disabled"
                              value="disabled"
                            >
                              Disabled
                            </option>
                          </select>
                        </div>
                      </FormGroup>
                    </div>
                  </Col>
                  <Col
                    bsClass="col"
                    className="mb-1"
                    componentClass="div"
                    sm={12}
                  >
                    <div
                      className="mb-1 col-sm-12"
                    >
                      <div
                        className="font-weight-bold py-1"
                      >
                        Snippets
                      </div>
                      <p>
                        Use code snippets in editor
                      </p>
                      <FormGroup
                        bsClass="form-group"
                        controlId="snippets"
                      >
                        <div
                          className="form-group"
                        >
                          <select
                            className="form-control snippets-control"
                            onChange={[Function]}
                            value="disabled"
                          >
                            <option
                              key="enabled"
                              value="enabled"
                            >
                              Enabled
                            </option>
                            <option
                              key="disabled"
                              value="disabled"
                            >
                              Disabled
                            </option>
                          </select>
                        </div>
                      </FormGroup>
                    </div>
                  </Col>
                </div>
              </Row>
            </div>
          </Grid>
        </div>
      </EditorSettings>
    </Connect(EditorSettings)>
  </SidePanel>
</Connect(SidePanel)>
`;
